base/sparse/linalg.jl

```diff
@@ -585,39 +585,41 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
         S_old = copy(S)
         for j = 1:t
             for i = 1:n
-                S[i,j] = Y[i,j]==0?1:sign(Y[i,j])
+                S[i,j] = Y[i,j]==0?1:(Y[i,j]/abs(Y[i,j]))
```

Comment:
Ah, I didn't realize sign was defined for complex numbers. The change then
boils down to the if statement :) I'll revert to the call to sign() and add
some tests.

On Wed, Aug 12, 2015, 6:17 PM Stefan Karpinski <notifications@github.com>
wrote:

> In base/sparse/linalg.jl
> <https://github.com/JuliaLang/julia/pull/12592#discussion_r36921191>:
>
> > @@ -585,39 +585,41 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
> >          S_old = copy(S)
> >          for j = 1:t
> >              for i = 1:n
> > -                S[i,j] = Y[i,j]==0?1:sign(Y[i,j])
> > +                S[i,j] = Y[i,j]==0?1:(Y[i,j]/abs(Y[i,j]))
>
> sign(z::Complex) is already defined this way, but is more efficient for
> non-complex types.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaLang/julia/pull/12592/files#r36921191>.
>